### ISSUE TITLE
Ability to send funds in Multiplayer

### ISSUE THREAD

**Issue by benhbell** [OpenRA/OpenRA#17735]
## Edit
This is talking about Dune2K, since CC has supply trucks already. Given the existence of supply trucks in CC, I have edited the proposed solution.

## Motivation
In many multiplayer team games the balance of funds vs units vs research is a bit imbalanced. some players are just better at economy than others, while others are better at war. Combining these strengths is part of what makes team playing so powerful. 

Players should be able to send funds to other players in multiplayer

## Proposed solution
A user should be able to send a supply truck to another user's base, when the supply truck arrives (construction yard?) the credits/spice can be added to their stores. If the user does not have enough storage/silos, then only the money the user is capable of receiving is received by the player. The rest of the money is lost.

## Side effects
Since there are no voice overs for this event, users might not be aware they got money. This is also a proposed change to team dynamics. The weakest link will not always be the player with the weakest military if they have a strong financial ally.

**Comment by abcdefg30** [comment]
Isn't that what supply trucks are for?

**Comment by dragunoff** [comment]
> 
> 
> Isn't that what supply trucks are for?

Supply trucks are an actual unit that has to traverse the map in order to deliver the funds. The OP is talking about transferring funds behind the scenes via the UI. This feature exists in some RTS games.

**Comment by AoAGeneral** [comment]
The biggest side effect from this is in StarCraft 2. Where people started doing money dump strategies and building scales of units in very short times (Old school marine rush memes). This would create a rather bad disaster in team games. The bigger the team games the worst it gets.

Supply trucks is the alternative to prevent this issue from coming up while still allowing cash sending to happen.

**Comment by pchote** [comment]
As mentioned, this idea was considered but then rejected in favour of the supply trucks in order to have an opportunity cost on the ability to transfer funds (loss of the vehicle production queue + risk of enemy interception).

**Comment by benhbell** [comment]
Ah the supply trucks aspect I great for cc, but there's no equivalent in openra dune.

**Comment by MustaphaTR** [comment]
In my mercenary missions and Chrono Fault, i used Turretless Combat Tanks as supply vehicles. if someone makes a fitting cameo, i think we can add them to the base mod.

**Comment by benhbell** [comment]
So perhaps we are just missing a unit for Dune that can send supplies (spice) to another player?

**Comment by MustaphaTR** [comment]
Yes, and i think Turretless Combat Tanks fit for this purpose as the unit artwork. But someone would need to make icons for the production UI for it.
In the custom missions i mentioned they are mission units that come out of the map, so i never needed an icon for them there.

**Comment by benhbell** [comment]
I can do the icon - what do I need to do to update the specs of this ticket to apply here?

**Comment by benhbell** [comment]
![MoneyTransport](https://user-images.githubusercontent.com/966644/125506992-c4941d68-7823-41c0-bf91-76953c4e0e9b.png)

I've gone ahead and done an icon if we want. First time making icons for this mod. If anyone knows how to more properly extract some assets from the data.r8 file or what the right file types and sizes I need it to be. 

Anything else I can do to see this feature implemented?

**Comment by benhbell** [comment]
Also adjusted the description of the issue.

**Comment by benhbell** [comment]
> Yes, and i think Turretless Combat Tanks fit for this purpose as the unit artwork. But someone would need to make icons for the production UI for it.
> In the custom missions i mentioned they are mission units that come out of the map, so i never needed an icon for them there.

Where can I find your turret-less tank for dune? @MustaphaTR

**Comment by MustaphaTR** [comment]
Turret and Body art of the Combat Tanks are already seperate, if you need the code you can check my maps, but they are a bit out of date: https://resource.openra.net/maps/33160/ https://resource.openra.net/maps/33159/ https://resource.openra.net/maps/41434/

**Comment by benhbell** [comment]
@dragunoff is there anything else that needs to be done to put this up for consideration by the team for openDune?

**Comment by dragunoff** [comment]
> @dragunoff is there anything else that needs to be done to put this up for consideration by the team for openDune?

@benhbell As far as I know the goal of the d2k mod for OpenRA is to be close to the original game. So I don't know if adding new features such as a supply truck is going to be considered before all of the core game features are completed. But as I'm not involved with d2k I'm not the right person who can answer that question properly.

**Comment by abcdefg30** [comment]
Fwiw, I'd be ok with adding this unit for Skirmish/MP.

**Comment by benhbell** [comment]
yea, hoping that since CC already has this unit/logic it is not a complete from the ground rewrite
